src/System.Private.Interpreter/src/Internal/Runtime/Interpreter/ILImporter.Interpreter.cs

```diff
@@ -239,22 +255,1021 @@ private void ImportLoadFloat(double value)
+                                result = (UIntPtr)value;
+                            }
+                            else
+                            {
+                                int value = stackItem.AsInt32();
+                                result = (UIntPtr)value;
+                            }
+                        }
+                        else if (stackItem.Kind == StackValueKind.Int64)
+                        {
+                            if (unsigned)
+                            {
+                                ulong value = (ulong)stackItem.AsInt64();
+                                result = (UIntPtr)value;
+                            }
+                            else
+                            {
+                                result = (UIntPtr)stackItem.AsInt64();
+                            }
+                        }
+                        else if (stackItem.Kind == StackValueKind.Float)
+                        {
+                            result = (UIntPtr)stackItem.AsDouble();
+                        }
+                        else if (stackItem.Kind == StackValueKind.NativeInt)
+                        {
+                            if (unsigned)
+                            {
+                                result = (UIntPtr)stackItem.AsNativeInt().ToPointer();
+                            }
+                            else
+                            {
+                                result = new UIntPtr(stackItem.AsNativeInt().ToPointer());
+                            }
+                        }
+                        else
+                        {
+                            // TODO: Figure out how the CLR converts an object to an integer representation
+                        }
+
+                        _interpreter.EvaluationStack.Push(StackItem.FromNativeInt(new IntPtr(result.ToPointer())));
+                        break;
+                    }
+                default:
+                    ThrowHelper.ThrowInvalidProgramException();
+                    break;
+            }
         }
 
         private void ImportUnaryOperation(ILOpcode opCode)
         {
-            throw new NotImplementedException();
+            StackItem stackItem = PopWithValidation();
+            switch (opCode)
+            {
+                case ILOpcode.neg:
+                    {
+                        if (stackItem.Kind == StackValueKind.Int32)
+                        {
+                            int value = stackItem.AsInt32();
+                            _interpreter.EvaluationStack.Push(StackItem.FromInt32(-value));
+                        }
+                        else if (stackItem.Kind == StackValueKind.Int64)
+                        {
+                            long value = stackItem.AsInt64();
+                            _interpreter.EvaluationStack.Push(StackItem.FromInt64(-value));
+                        }
```

Comment:
This should also handle `StackValueKind.NativeInt`

Reply:
Done